tools/operator/launcher/k8s/network.go

```diff
@@ -2,241 +2,522 @@ package k8s
+func (k8s K8s) resources(component string, resource networkspec.Resource) v12.ResourceRequirements {
+	limitsCPU := apiResource.MustParse(resource.Limits.CPU)
+	limitsMemory := apiResource.MustParse(resource.Limits.Memory)
+	requestsCPU := apiResource.MustParse(resource.Requests.CPU)
+	requestsMemory := apiResource.MustParse(resource.Requests.Memory)
+	limitsMap := make(map[v12.ResourceName]apiResource.Quantity)
+	requestsMap := make(map[v12.ResourceName]apiResource.Quantity)
+	limitsMap[v12.ResourceName("cpu")] = apiResource.Quantity(limitsCPU)
+	limitsMap[v12.ResourceName("memory")] = apiResource.Quantity(limitsMemory)
+	requestsMap[v12.ResourceName("cpu")] = apiResource.Quantity(requestsCPU)
+	requestsMap[v12.ResourceName("memory")] = apiResource.Quantity(requestsMemory)
+	resources := v12.ResourceRequirements{Limits: v12.ResourceList(limitsMap), Requests: v12.ResourceList(requestsMap)}
+	return resources
+}
 
-	kubeConfigPath := fmt.Sprintf("--kubeconfig=%s", k.KubeConfigPath)
-	args := []string{kubeConfigPath, k.Action, k8sType}
-	if k8sType == "secret" {
-		args = append(args, "generic")
+func (k8s K8s) caLaunch(id int, orgName, caImage string) LaunchConfig {
+
+	l := LaunchConfig{Name: fmt.Sprintf("ca%d-%s", id, orgName), Type: "ca"}
+	containers := make([]v12.Container, 0)
+	container := v12.Container{Name: "ca", Command: []string{"fabric-ca-server"}, Args: []string{"start", "-b", "admin:adminpw", "-d"}}
+	container.Image = caImage
+	envs := make([]v12.EnvVar, 0)
+	env := v12.EnvVar{Name: "FABRIC_CA_HOME", Value: "/etc/hyperledger/fabric-ca-server"}
+	envs = append(envs, env)
+	env = v12.EnvVar{Name: "FABRIC_CA_SERVER_CA_KEYFILE", Value: "/etc/hyperledger/fabric/artifacts/ca-priv_sk"}
+	envs = append(envs, env)
+	env = v12.EnvVar{Name: "FABRIC_CA_SERVER_CA_CERTFILE", Value: fmt.Sprintf("/etc/hyperledger/fabric/artifacts/ca.%s-cert.pem", orgName)}
+	envs = append(envs, env)
+	env = v12.EnvVar{Name: "FABRIC_CA_SERVER_TLS_ENABLED", Value: "true"}
+	envs = append(envs, env)
+	env = v12.EnvVar{Name: "FABRIC_CA_SERVER_TLS_KEYFILE", Value: "/etc/hyperledger/fabric/artifacts/tlsca-priv_sk"}
+	envs = append(envs, env)
+	env = v12.EnvVar{Name: "FABRIC_CA_SERVER_TLS_CERTFILE", Value: fmt.Sprintf("/etc/hyperledger/fabric/artifacts/tlsca.%s-cert.pem", orgName)}
+	envs = append(envs, env)
+	env = v12.EnvVar{Name: "FABRIC_CA_SERVER_CA_NAME", Value: fmt.Sprintf("ca%d-%s", id, orgName)}
+	envs = append(envs, env)
+	container.Env = envs
+	volumeMount := v12.VolumeMount{MountPath: "/etc/hyperledger/fabric/artifacts/", Name: "cacerts"}
+	container.VolumeMounts = append(container.VolumeMounts, volumeMount)
+	containers = append(containers, container)
+	l.Containers = containers
+	volume := v12.Volume{Name: "cacerts", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-ca", orgName)}}}}
+	l.Volumes = append(l.Volumes, volume)
+	return l
+}
```

Comment:
We can greatly improve the readability of this function:

```
func (k8s K8s) caLaunchConfig(id int, orgName, caImage string) LaunchConfig {
	container := corev1.Container{
		Args:    []string{"start", "-b", "admin:adminpw", "-d"},
		Command: []string{"fabric-ca-server"},
		Env: []corev1.EnvVar{
			{Name: "FABRIC_CA_HOME", Value: "/etc/hyperledger/fabric-ca-server"},
			{Name: "FABRIC_CA_SERVER_CA_KEYFILE", Value: "/etc/hyperledger/fabric/artifacts/ca-priv_sk"},
			{Name: "FABRIC_CA_SERVER_CA_CERTFILE", Value: fmt.Sprintf("/etc/hyperledger/fabric/artifacts/ca.%s-cert.pem", orgName)},
			{Name: "FABRIC_CA_SERVER_TLS_ENABLED", Value: "true"},
			{Name: "FABRIC_CA_SERVER_TLS_KEYFILE", Value: "/etc/hyperledger/fabric/artifacts/tlsca-priv_sk"},
			{Name: "FABRIC_CA_SERVER_TLS_CERTFILE", Value: fmt.Sprintf("/etc/hyperledger/fabric/artifacts/tlsca.%s-cert.pem", orgName)},
			{Name: "FABRIC_CA_SERVER_CA_NAME", Value: fmt.Sprintf("ca%d-%s", id, orgName)},
		},
		Image: caImage,
		Name:  "ca",
		VolumeMounts: []corev1.VolumeMount{{
			MountPath: "/etc/hyperledger/fabric/artifacts/",
			Name:      "cacerts"},
		},
	}

	volume := corev1.Volume{
		Name: "cacerts",
		VolumeSource: corev1.VolumeSource{
			ConfigMap: &corev1.ConfigMapVolumeSource{
				LocalObjectReference: corev1.LocalObjectReference{
					Name: fmt.Sprintf("%s-ca", orgName),
				},
			},
		},
	}

	launchConfig := LaunchConfig{
		Containers: []corev1.Container{container},
		Name:       fmt.Sprintf("ca%d-%s", id, orgName),
		Type:       "ca",
		Volumes:    []corev1.Volume{volume},
	}
	return launchConfig
}
```

Reply:
Yes I agree.. I also felt to change to something with more readability. Thank you. Fixed.